Bottom: 97f723db8e4311dfb8ce79850eb36e8e7c6ef964
Top:    50629965649e1c492b0cf8ba9ee5cb2501e46aef
Author: Jon Smirl <jonsmirl@gmail.com>
Date:   2009-05-26 12:18:03 -0400

Refresh of timur

---

diff --git a/arch/powerpc/include/asm/delay.h b/arch/powerpc/include/asm/delay.h
index af4a270..d0e68d3 100644
--- a/arch/powerpc/include/asm/delay.h
+++ b/arch/powerpc/include/asm/delay.h
@@ -52,16 +52,18 @@ extern void udelay(unsigned long usecs);
  *
  * gcc will optimize out the if-statement if @delay is a constant.
  */
-#define spin_event_timeout(condition, timeout, delay, rc)                   \
-{                                                                           \
-	unsigned long __loops = tb_ticks_per_usec * timeout;                \
-	unsigned long __start = get_tbl();                                  \
-	while (!(rc = (condition)) && (tb_ticks_since(__start) <= __loops)) \
-		if (delay)                                                  \
-			udelay(delay);                                      \
-		else	                                                    \
-			cpu_relax();                                        \
-}
+#define spin_event_timeout(condition, timeout, delay)                         \
+({                                                                            \
+       typeof(condition) __ret;                                               \
+       unsigned long __loops = tb_ticks_per_usec * timeout;                   \
+       unsigned long __start = get_tbl();                                     \
+       while (!(__ret = (condition)) && (tb_ticks_since(__start) <= __loops)) \
+               if (delay)                                                     \
+                       udelay(delay);                                         \
+               else                                                           \
+                       cpu_relax();                                           \
+       __ret;                                                                 \
+})
 
 #endif /* __KERNEL__ */
 #endif /* _ASM_POWERPC_DELAY_H */
